flatgfa/src/packedseq.rs

```diff
@@ -204,20 +285,46 @@ pub fn create_slice(vec: &PackedVec, s: std::ops::Range<usize>) -> PackedSlice<'
 
 /// Returns a vector containing the base pairs referenced by `slice`
 pub fn get_slice_seq(slice: PackedSlice<'_>) -> Vec<Nucleotide> {
-    slice.vec_ref.get_range(slice.span)
+    slice.vec_ref.as_ref().get_range(slice.span)
+}
+
+pub fn total_bytes(seq: &PackedSeqView) -> usize {
+    let num_elems = seq.len();
+    let seq_size = if num_elems % 2 == 1 {
+        (num_elems / 2) + 1
+    } else {
+        num_elems / 2
+    };
+
+    let toc_size = std::mem::size_of::<PackedToc>();
+    toc_size + seq_size
+}
+
+pub fn import(filename: &str) -> Vec<Nucleotide> {
+    let mmap = memfile::map_file(filename); // args[2] is the filename
+    PackedSeqView::view(&mmap).get_elements()
+}
+
+pub fn export(seq: PackedSeqView, filename: &str) {
+    let num_bytes = total_bytes(&seq);
+    let mut mem = map_new_file(filename, num_bytes as u64);
+    let mut buf = vec![0u8; num_bytes];
+    let buf_ref = &mut buf;
+    PackedSeqView::dump(&seq, buf_ref);
+    mem[..buf_ref.len()].copy_from_slice(buf_ref);
```

Comment:
This entails two copies where one would suffice:

* It allocates a vector and uses `dump` to put all the data into that vector.
* It then copies all that data into `mem`.

Instead, let's just `dump` the data directly into `mem`. Something roughly like this should suffice:

```rs
seq.dump(mem);
```